okhttp/third_party/okhttp/java/io/grpc/okhttp/internal/Platform.java

```diff
@@ -105,47 +119,71 @@ public void connectSocket(Socket socket, InetSocketAddress address,
   }
 
   /** Attempt to match the host runtime to a capable Platform implementation. */
+  @SuppressWarnings("unchecked")
   private static Platform findPlatform() {
-    // Attempt to find Android 2.3+ APIs.
+    // Find the conscrypt security provider if we can
+    Class<Provider> conscryptProviderClass = null;
     try {
+      conscryptProviderClass =
+          (Class<Provider>) Class.forName("org.conscrypt.OpenSSLProvider");
+    } catch (Throwable t) {
```

Comment:
I'm a little wary of Android here and the fact that there are so many permutations of conscrypt X platforms possible that this may end up being more common than we would like

OTOH OkHttp is using CNFE for this and it's probably fairly well validated so perhaps its just fine.

Done